Tools/Schema.NET.Tool/SchemaSourceGenerator.cs

```diff
@@ -0,0 +1,198 @@
+namespace Schema.NET.Tool
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Schema.NET.Tool.CustomOverrides;
+    using Schema.NET.Tool.GeneratorModels;
+    using Schema.NET.Tool.Repositories;
+    using Schema.NET.Tool.Services;
+
+    [Generator]
+    public class SchemaSourceGenerator : ISourceGenerator
+    {
+        private IEnumerable<GeneratorSchemaObject> SchemaObjects { get; set; }
+
+        public void Initialize(GeneratorInitializationContext context)
+        {
+#pragma warning disable IDE0022 // Use expression body for methods
+            Task.Run(async () =>
```

Comment:
Why do we wrap this code in `Task.Run` and then synchronously wait for it?

Reply:
A combination of wanting to use the async/await pattern for the actual main logic while also trying to avoid potential deadlocks. Using `Task.Run` pushes it to the thread pool whereas async doesn't. I can give it another crack to remove/simplify it - this just happened to be what I had when I first got the source generator running at all.